### ISSUE TITLE
PackageCompiler requires 7z

### ISSUE THREAD

**Issue by bgeihe** [trixi-framework/libtrixi#142]
If you do not have it installed, you might see something like:
```
  Downloaded artifact: MKL
  Downloaded artifact: MKL
fatal: error thrown and no exception handler available.
InitError(mod=:MKL_jll, error=ErrorException("Unable to automatically download/install artifact 'MKL' from sources listed in '/home/bene/.julia/packages/MKL_jll/8Hu7G/Artifacts.toml'.
Sources attempted:
- https://pkg.julialang.org/artifact/d670351e2fcdac07d02cf73bda5f53e9bea796a6
    Error: IOError: could not spawn setenv(`7z x /tmp/jl_KNK5I7o7of-download.gz -so`,...
```

**Comment by sloede** [comment]
Where was this reported?

**Comment by bgeihe** [comment]
I am sure that things had worked at some point on the same machine without 7z.

**Comment by sloede** [comment]
Is this Roci?

**Comment by bgeihe** [comment]
No, this was on my machine.
I would assume roci and the github runners have 7z installed.

**Comment by bgeihe** [comment]
It seems strange to me, though, because before I tried PackageCompiler (and installed p7zip) I ran `libtrixi-init-julia`, which reported
```
 MKL_jll [856f044c-d86e-5d09-b602-aeab76dc8ba7]
│   Downloading artifact: MKL
└
```
without errors.

**Comment by sloede** [comment]
Yes, PackageCompiler does not bundle the 7z library, while Julia itself does. Maybe it needs to be included in the libraries that are bundled by PC with libraries/apps - I had a discussion with Kristoffer Carlsson of JuliaHub about this on Slack today.

**Comment by bgeihe** [comment]
Just to document what I tried:

- libtrixi on clean main branch
- `cd LibTrixi.jl/lib`
- `JULIA_PKG_DEVDIR=./dev julia --project=. -e 'import Pkg; Pkg.develop("PackageCompiler")'`
- `cd dev/PackageCompiler`
- `git switch msl/bundle-libexec`
- `cd ../..`
- set path to `libtrixi-julia` project directory in `Makefile`
- `make`
- `ls -l build/libexec/julia/7z` (exists!)
- `cd ../../examples`
- `make -f MakefileCompiled LIBTRIXI_PREFIX=$PWD/../LibTrixi.jl/lib/build`
- `JULIA_DEPOT_PATH=./non-exisiting ./trixi_controller_simple_c ./non-existing ../LibTrixi.jl/examples/libelixir_tree2d_dgsem_advection_amr.jl`
- ```
  *** Trixi controller ***   Initialize Trixi
    Downloaded artifact: MKL
    Downloaded artifact: MKL
  fatal: error thrown and no exception handler available.
  InitError(mod=:MKL_jll, error=ErrorException("Unable to automatically download/install artifact 'MKL' from sources listed in '/home/bgeihe/.julia/packages/MKL_jll/8Hu7G/Artifacts.toml'.
